### PR TITLE
feat(Python3): Build support for Python3.6

### PR DESCRIPTION
**Pull request by jfuss**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jfuss** [comment]
I couldn't seem to get `make func-test` and `make integ-test` to work properly on travis and wasn't really going anywhere. So for now, we only run `make pr` (linting, unit tests) when PRs are created. We do need to expand this to func and integ but didn't want to go down a deep hole while trying to wrap up Py3 stuff.

If you have a better idea on something else I should be doing to make these work, I am happy to address and give it a try.

File: README.rst [hunk 1]
```diff
@@ -101,7 +101,7 @@ Prerequisites
 ~~~~~~~~~~~~~
 
 - Docker
-- Python2.7 (Python3+ not yet supported)
+- Python2.7 or Python3.6
```
Comment (sanathkr) [inline README.rst]: omg yaay 🎉

File: README.rst [hunk 2]
```diff
@@ -253,7 +253,7 @@ homebrew and try again:
 
 .. code:: bash
 
-   brew install python@2
+   brew install python@3
```
Comment (sanathkr) [inline README.rst]: remove the version identifier here.

File: samcli/commands/validate/lib/sam_template_validator.py [hunk 3]
```diff
@@ -56,9 +55,28 @@ def is_valid(self):
...
+        # does not match the schema. The logger they use is the root logger instead of one scoped to
+        # their module. Currently this does not cause templates to fail, so we will suppress this
+        # by patching the logging.warning method that is used in that class.
+        class WarningSuppressLogger(object):
+
+            def __init__(self, obj_to_patch):
+                self.obj_to_patch = obj_to_patch
+
+            def __enter__(self):
+                self.obj_to_patch.warning = self.warning
```
Comment (sanathkr) [inline samcli/commands/validate/lib/sam_template_validator.py]: wow! Something to live with until SAM Translator makes a new release.

File: tests/functional/local/apigw/test_service.py [hunk 4]
```diff
@@ -16,6 +17,8 @@
 from samcli.commands.local.lib.local_lambda import LocalLambdaRunner
 from samcli.local.docker.manager import ContainerManager
 
+logging.getLogger().setLevel(logging.DEBUG)
```
Comment (sanathkr) [inline tests/functional/local/apigw/test_service.py]: Can you remove this line?

File: tox.ini [hunk 5]
```diff
@@ -4,8 +4,10 @@
...
 [tox]
-envlist = py27
+envlist = py27, py36
 
 [testenv]
 whitelist_externals = make
-commands = make pr
+commands =
+    make pr
+passenv = *
```
Comment (sanathkr) [inline tox.ini]: does this work?

File: README.rst (hunk 2 above)
Comment (jfuss) [reply README.rst]: done.

**Pull request merged**